drivers/flash/flash_native_posix.c

```diff
@@ -1,233 +0,0 @@
-/*
- * Copyright (c) 2019 Jan Van Winkel <jan.van_winkel@dxplore.eu>
- *
- * SPDX-License-Identifier: Apache-2.0
- */
-
-#include <device.h>
-#include <drivers/flash.h>
-
-#include <unistd.h>
-#include <sys/types.h>
-#include <sys/stat.h>
-#include <sys/mman.h>
-#include <fcntl.h>
-#include <string.h>
-#include <errno.h>
-
-#include "cmdline.h"
-#include "soc.h"
-
-#define LOG_LEVEL CONFIG_FLASH_LOG_LEVEL
-#include <logging/log.h>
-LOG_MODULE_REGISTER(flash_native_posix);
-
-static const char default_flash_path[] = "flash.bin";
-
-struct flash_native_posix_data {
-	const char *flash_path;
-	int fd;
-	u8_t *flash;
-	bool init_called;
-};
-
-struct flash_native_posix_config {
-	size_t flash_size;
-#if defined(CONFIG_FLASH_PAGE_LAYOUT)
-	struct flash_pages_layout layout;
-#endif
-};
-
-#define DEV_NAME(dev) ((dev)->config->name)
-#define DEV_CONFIG(dev) ((dev)->config->config_info)
-#define DEV_DATA(dev) \
-	((struct flash_native_posix_data *const)(dev)->driver_data)
-
-static int flash_native_posix_read(struct device *dev, off_t offset, void *data,
-				   size_t size)
-{
-	struct flash_native_posix_data *const dev_data = DEV_DATA(dev);
-	const struct flash_native_posix_config *config = DEV_CONFIG(dev);
-
-	if (dev_data->flash == MAP_FAILED) {
```

Comment:
Now you have given up on this checks.. I wonder if the warnings during init should become errors instead.

Reply:
As per the discussion in the EPROM one, as it is now it is fine